subsys/net/lib/download_client/src/download_client.c

```diff
@@ -759,17 +764,6 @@ int download_client_start(struct download_client *client, const char *file,
 	client->offset = 0;
 	client->http.has_header = false;
 
-	if (client->proto == IPPROTO_UDP || client->proto == IPPROTO_DTLS_1_2) {
-		if (IS_ENABLED(CONFIG_COAP)) {
-			coap_block_init(client, from);
-		}
-	}
-
-	err = request_send(client);
-	if (err) {
-		return err;
-	}
-
 	LOG_INF("Downloading: %s [%u]", client->file, client->progress);
```

Comment:
Maybe move the log line as well?

Reply:
This does not matter. It kind of starts the request here.